### PR TITLE
Migrating the Azpipeline templates to Go templates

### PR DESCRIPTION
**Pull request by Vidya2606**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a  breaking change which will impact consuming tool(s).

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/azurePipelines/azurePipelines.go [hunk 1]
```diff
@@ -125,7 +125,7 @@ func (p *AzurePipelines) CreatePipelineFiles(deployType string, draftConfig *con
 		return fmt.Errorf("error applying default variables: %w", err)
 	}
 
-	if err := osutil.CopyDir(p.pipelineTemplates, srcDir, p.dest, draftConfig, templateWriter); err != nil {
+	if err := osutil.CopyDirWithTemplates(p.pipelineTemplates, srcDir, p.dest, draftConfig, templateWriter); err != nil {
```
Comment (OliverMKing) [inline pkg/azurePipelines/azurePipelines.go:128]: we should have some kind of unit test proving that these templates are properly filled out with the right Draft variables. Do we have that?

I like the fixture pattern

Comment (Vidya2606) [reply pkg/azurePipelines/azurePipelines.go:128]: I have tested them manually and yes, all the variables are replaced correctly!

Comment (OliverMKing) [reply pkg/azurePipelines/azurePipelines.go:128]: we need a unit test to prove and ensure that

Comment (Vidya2606) [reply pkg/azurePipelines/azurePipelines.go:128]: I have updated the unit tests with the fixtures!

File: pkg/azurePipelines/azurePipelines_test.go [hunk 2]
```diff
@@ -148,7 +162,7 @@ func newDraftConfig() *config.DraftConfig {
 			{
 				Name: "MANIFESTPATH",
 				Default: config.BuilderVarDefault{
-					Value: "manifests",
+					Value: "test/manifests",
```
Comment (bosesuneha) [inline pkg/azurePipelines/azurePipelines_test.go:165]: any reason why `test` dir was added?

Comment (Vidya2606) [reply pkg/azurePipelines/azurePipelines_test.go:165]: It was added to match the manifests path in the fixtures directory.

**Review by bfoley13 — APPROVED** [review]

**Review by bosesuneha — APPROVED** [review]

**Pull request merged**